### ISSUE TITLE
[Craft 5] Invalid Owner ID error in channel

### ISSUE THREAD

**Issue by TheFunkyMonk** [internetztube/craft-element-relations#35]
This one is kind of hard to define repro steps for, but I'm glad to provide a db backup and composer files via email if it's helpful.

I created an Element Relations field and added it to the field layout of several channels. In most channels, it's working as expected, but in one specific channel I don't see any significant difference between, I'm getting an "Invalid Owner ID" error when trying to view the entry's relations field.

```
[web.ERROR] [yii\base\InvalidConfigException] yii\base\InvalidConfigException: Invalid owner ID: 3746 in /var/www/html/vendor/craftcms/cms/src/base/NestedElementTrait.php:158
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/base/NestedElementTrait.php(203): craft\elements\Entry->getPrimaryOwner()
#1 /var/www/html/vendor/craftcms/cms/src/base/NestedElementTrait.php(235): craft\elements\Entry->getOwner()
#2 /var/www/html/vendor/craftcms/cms/src/base/Element.php(3294): craft\elements\Entry->getField()
#3 /var/www/html/vendor/craftcms/cms/src/elements/Entry.php(1646): craft\base\Element->canView(Object(craft\elements\User))
#4 /var/www/html/vendor/craftcms/cms/src/services/Elements.php(4183): craft\elements\Entry->canView(Object(craft\elements\User))
#5 /var/www/html/vendor/craftcms/cms/src/helpers/Cp.php(811): craft\services\Elements->canView(Object(craft\elements\Entry), Object(craft\elements\User))
#6 /var/www/html/vendor/craftcms/cms/src/helpers/Cp.php(470): craft\helpers\Cp::baseElementAttributes(Object(craft\elements\Entry), Array)
#7 /var/www/html/vendor/craftcms/cms/src/helpers/Cp.php(1017): craft\helpers\Cp::elementChipHtml(Object(craft\elements\Entry), Array)
#8 /var/www/html/vendor/internetztube/craft-element-relations/src/twigextensions/ControlPanel.php(15): craft\helpers\Cp::elementPreviewHtml(Array)
#9 /var/www/html/storage/runtime/compiled_templates/fe/fe602748cbd630f83e73747b7fa1ea95.php(71): internetztube\elementRelations\twigextensions\ControlPanel->internetztube\elementRelations\twigextensions\{closure}(Array)
#10 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_8ec2633b68b841f22d01a657276f2743->doDisplay(Array, Array)
#11 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#12 /var/www/html/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
#13 /var/www/html/vendor/twig/twig/src/TemplateWrapper.php(38): Twig\Template->render(Array)
#14 /var/www/html/vendor/twig/twig/src/Environment.php(280): Twig\TemplateWrapper->render(Array)
#15 /var/www/html/vendor/craftcms/cms/src/web/View.php(494): Twig\Environment->render('element-relatio...', Array)
#16 /var/www/html/vendor/internetztube/craft-element-relations/src/controllers/ElementRelationsController.php(27): craft\web\View->renderTemplate('element-relatio...', Array)
#17 [internal function]: internetztube\elementRelations\controllers\ElementRelationsController->actionGetByElementId()
#18 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#19 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#20 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('get-by-element-...', Array)
#21 /var/www/html/vendor/craftcms/cms/src/web/Application.php(349): yii\base\Module->runAction('element-relatio...', Array)
#22 /var/www/html/vendor/craftcms/cms/src/web/Application.php(650): craft\web\Application->runAction('element-relatio...', Array)
#23 /var/www/html/vendor/craftcms/cms/src/web/Application.php(311): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#24 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#25 /var/www/html/web/index.php(12): yii\base\Application->run()
#26 {main} {"memory":5013552,"exception":"[object] (yii\\base\\InvalidConfigException(code: 0): Invalid owner ID: 3746 at /var/www/html/vendor/craftcms/cms/src/base/NestedElementTrait.php:158)"} 
```

Craft CMS version: 5.2.1
Element Relations version: 3.0.2.2
PHP version: 8.2.19

**Comment by TheFunkyMonk** [comment]
Actually, I think I can provide some more info here. I just realized this happens when the related entry is referencing the entry in question inside a Matrix field that's on the field layout of a Neo block.

![2024-06-18 at 10 42 39](https://github.com/internetztube/craft-element-relations/assets/1728770/103b6357-a670-4cdb-8299-79b620ef6ca1)

**Comment by internetztube** [comment]
Ok, I'll try to reproduce it.

**Comment by internetztube** [comment]
I tried to reproduce this issue but was unable to. Would you be able to send your `composer.json`, `composer.lock` files, and database export to support@internetztube.net and point to the entry where you see this error triggered?

**Comment by TheFunkyMonk** [comment]
Sure thing! Thanks for the quick response.

**Comment by internetztube** [comment]
Should be fixed with `3.0.3`. Thanks for reporting! ☺️
```
composer update internetztube/craft-element-relations
```

**Comment by TheFunkyMonk** [comment]
Quickest fix ever, appreciate it!
